include/oneapi/dpl/pstl/hetero/dpcpp/unseq_backend_sycl.h

```diff
@@ -586,7 +585,8 @@ struct __scan
         __init_processing<_Tp> __use_init{};
 
         ::std::size_t __shift = 0;
-        __internal::__invoke_if_not(_Inclusive{}, [&]() { __shift = 1; });
+        if (!_Inclusive::value)
```

Comment:
const ::std::size_t __shift = _Inclusive::value ? 1 : 0;

Reply:
Or why not ```if constexpr (!_Inclusive::value)``` like below ?

Reply:
Or like in some other places already wrote:
```
constexpr auto __shift = _Inclusive{} ? .....;
```

Reply:
Good catch. I forgot to add the `constexpr` here. I will correct that.